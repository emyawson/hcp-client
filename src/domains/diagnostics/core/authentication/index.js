export {
  validateSession,
  signOut,
  onRedirectToLogin,
} from './authentication.actions';
export {
  getToken,
  getEc6Token,
  getIsValidSession,
  selectHasTemporaryPassword,
  selectHasLoggedOut,
} from './authentication.selectors';
export {
  LOGIN,
  LOGIN_BASE,
  FIRST_TIME,
  FIRST_TIME_BASE,
  VALIDATE_SESSION,
  VALIDATE_SESSION_BASE,
  RESET_PASSWORD,
  RESET_PASSWORD_BASE,
  SIGN_OUT,
  SIGN_OUT_BASE,
  REDIRECT_TO_LOGIN,
} from './authentication.constants';
