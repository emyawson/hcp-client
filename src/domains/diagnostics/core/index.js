export { validateSession } from './authentication';
export { createModal, MODAL_TYPES } from './modal';
export {
  fetchPatientRequest,
  selectPatient,
  selectPatientDevices,
} from './patient';
export {
  clearPatientDateRange,
  fetchPatientDateRangeRequest,
  onPatientDateRangeChange,
  selectPatientFirstMeasurementDate,
  selectPatientLastMeasurementDate,
  selectPatientStartDate,
  selectPatientEndDate,
} from './patient-date-range';
export {
  createRequestActions,
  createRequestActionTypes,
  REQUEST,
  REQUEST_ANIMATION_DELAY,
  selectOngoingRequests,
} from './request';
export {
  getThresholdsRequest,
  getTimeIntervalsRequest,
  getTrafficLightColorFromStatus,
  TRAFFIC_LIGHT_STATES,
} from './strip-delivery';
