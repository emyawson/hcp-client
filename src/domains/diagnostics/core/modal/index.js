export { createModal } from './modal.actions';
export { MODAL_ACTIONS, MODAL_TYPES } from './modal.constants';
