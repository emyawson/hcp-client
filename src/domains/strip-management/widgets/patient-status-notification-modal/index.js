export * from './patient-status-notification-modal.container';
export * from './patient-status-notification-modal.component';
