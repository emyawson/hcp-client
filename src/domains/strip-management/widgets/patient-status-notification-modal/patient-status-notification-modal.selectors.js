import { createStructuredSelector } from 'reselect';

import {
  selectLastTrafficLightStatus,
  selectLastTrafficLightStatusDateCalculatedFormatted,
  selectLastTrafficLightStatusConditions,
  selectLastTrafficLightStatusComment,
  selectHasLastTrafficLightStatusWithAlert,
} from 'src/core/strip-delivery';

export const PatientStatusNotificationModalConnector = createStructuredSelector(
  {
    lastTrafficLightStatus: selectLastTrafficLightStatus,
    lastTrafficLightStatusDateCalculated: selectLastTrafficLightStatusDateCalculatedFormatted,
    lastTrafficLightStatusConditions: selectLastTrafficLightStatusConditions,
    lastTrafficLightStatusComment: selectLastTrafficLightStatusComment,
    hasLastTrafficLightStatusWithAlert: selectHasLastTrafficLightStatusWithAlert,
  },
);
