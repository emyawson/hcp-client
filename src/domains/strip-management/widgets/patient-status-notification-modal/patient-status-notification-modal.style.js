import styled from 'styled-components';

export const PatientStatusNotificationModalBodyDiv = styled.div`
  background-color: ${props => props.theme.colors.white};
  padding: ${props => props.theme.spacing.four};
`;

export const PatientStatusNotificationButtonDiv = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  padding: ${props => props.theme.spacing.three};
`;

export const PatientStatusCommentWrapperDiv = styled.div`
  margin: 0 auto ${props => props.theme.spacing.three};
`;
