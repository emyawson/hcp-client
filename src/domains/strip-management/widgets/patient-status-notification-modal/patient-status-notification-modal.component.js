import React from 'react';

import {
  CursorBlock,
  LocalizedText,
  GridContainer,
  Button,
  Comment,
} from 'src/components';
import { CommonHeader, ModalBody } from 'src/widgets/modal';
import { StripStatusDetails } from 'src/domains/strip-management/components';
import { RenderIf, hasValue } from 'src/utils';
import { translate } from 'src/i18n';
import { XIcon } from 'src/assets/icons';
import { colors } from 'src/core/styles';

import {
  PatientStatusNotificationModalBodyDiv,
  PatientStatusNotificationButtonDiv,
  PatientStatusCommentWrapperDiv,
} from './patient-status-notification-modal.style';

import { ExpandedLastPatientStripStatusContainer } from '../patient-strip-status';

export const PatientStatusNotificationModal = ({
  lastTrafficLightStatusConditions,
  lastTrafficLightStatus,
  lastTrafficLightStatusComment,
  lastTrafficLightStatusDateCalculated,
  hasLastTrafficLightStatusWithAlert,
  destroyModal,
}) => (
  <ModalBody minWidth={960}>
    <CommonHeader>
      <LocalizedText
        fontSize={4}
        textKey={'modals.patientStatusNotification.title'}
      />
      <CursorBlock onClick={destroyModal}>
        <span>
          <XIcon height={14} fillColor={colors.white} />
        </span>
      </CursorBlock>
    </CommonHeader>
    <PatientStatusNotificationModalBodyDiv>
      <GridContainer>
        <ExpandedLastPatientStripStatusContainer
          displayOnly
          hideNextDeliveryDate
        >
          <StripStatusDetails
            conditions={lastTrafficLightStatusConditions}
            trafficLightStatus={lastTrafficLightStatus}
            hasLastDeliveryWithAlert={hasLastTrafficLightStatusWithAlert}
          />
          <RenderIf validate={hasValue(lastTrafficLightStatusComment)}>
            <PatientStatusCommentWrapperDiv>
              <Comment
                message={lastTrafficLightStatusComment}
                title={translate('stripDelivery.comments.adminComments.title')}
                date={lastTrafficLightStatusDateCalculated}
              />
            </PatientStatusCommentWrapperDiv>
          </RenderIf>
          <PatientStatusNotificationButtonDiv>
            <Button
              label={translate(
                'modals.patientStatusNotification.acknowledgePatientStatus',
              )}
              onClick={destroyModal}
            />
          </PatientStatusNotificationButtonDiv>
        </ExpandedLastPatientStripStatusContainer>
      </GridContainer>
    </PatientStatusNotificationModalBodyDiv>
  </ModalBody>
);
