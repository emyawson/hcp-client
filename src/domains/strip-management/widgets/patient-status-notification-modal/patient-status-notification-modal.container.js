import { connect } from 'react-redux';

import { PatientStatusNotificationModal } from './patient-status-notification-modal.component';
import { PatientStatusNotificationModalConnector } from './patient-status-notification-modal.selectors';

export const PatientStatusNotificationModalContainer = connect(
  PatientStatusNotificationModalConnector,
)(PatientStatusNotificationModal);
