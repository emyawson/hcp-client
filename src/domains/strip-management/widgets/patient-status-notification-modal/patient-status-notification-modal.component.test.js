import React from 'react';
import { shallow } from 'enzyme';

import { TRAFFIC_LIGHT_STATES } from 'src/core';

import { PatientStatusNotificationModal } from './patient-status-notification-modal.component';

describe('Patient status notification modal component', () => {
  const mockProps = {
    lastTrafficLightStatus: TRAFFIC_LIGHT_STATES.DELIVER,
    lastTrafficLightStatusConditions: {},
    lastTrafficLightStatusComment: 'comment',
    lastTrafficLightStatusDateCalculated: '2018-02-27T00:00:00.000Z',
    hasLastTrafficLightStatusWithAlert: false,
    destroyModal: () => {},
  };
  it('Renders properly', () => {
    const wrapper = shallow(<PatientStatusNotificationModal {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
