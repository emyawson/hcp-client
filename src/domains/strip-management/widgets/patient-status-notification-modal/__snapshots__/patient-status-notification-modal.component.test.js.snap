// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient status notification modal component Renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    destroyModal={[Function]}
    hasLastTrafficLightStatusWithAlert={false}
    lastTrafficLightStatus="DELIVER"
    lastTrafficLightStatusComment="comment"
    lastTrafficLightStatusConditions={Object {}}
    lastTrafficLightStatusDateCalculated="2018-02-27T00:00:00.000Z"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <LocalizedText
            fontSize={4}
            textKey="modals.patientStatusNotification.title"
          />
          <Styled(Styled(Component))
            onClick={[Function]}
          >
            <span>
              <Unknown
                fillColor="#FFF"
                height={14}
              />
            </span>
          </Styled(Styled(Component))>
        </styled.div>,
        <styled.div>
          <GridContainer>
            <Connect(Component)
              displayOnly={true}
              hideNextDeliveryDate={true}
            >
              <Unknown
                conditions={Object {}}
                hasLastDeliveryWithAlert={false}
                trafficLightStatus="DELIVER"
              />
              <f1
                validate={true}
              >
                <styled.div>
                  <Unknown
                    date="2018-02-27T00:00:00.000Z"
                    message="comment"
                    title="Administrative Comments"
                  />
                </styled.div>
              </f1>
              <styled.div>
                <Button
                  accessibilityLabel=""
                  buttonStyle="primary"
                  disabled={false}
                  fontSize="0.875rem"
                  label="Acknowledge Patient Status"
                  loading={false}
                  minWidth=""
                  onClick={[Function]}
                />
              </styled.div>
            </Connect(Component)>
          </GridContainer>
        </styled.div>,
      ],
      "minWidth": 960,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <LocalizedText
              fontSize={4}
              textKey="modals.patientStatusNotification.title"
            />,
            <Styled(Styled(Component))
              onClick={[Function]}
            >
              <span>
                <Unknown
                  fillColor="#FFF"
                  height={14}
                />
              </span>
            </Styled(Styled(Component))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "fontSize": 4,
              "textKey": "modals.patientStatusNotification.title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                <Unknown
                  fillColor="#FFF"
                  height={14}
                />
              </span>,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown
                  fillColor="#FFF"
                  height={14}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "fillColor": "#FFF",
                  "height": 14,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <GridContainer>
            <Connect(Component)
              displayOnly={true}
              hideNextDeliveryDate={true}
            >
              <Unknown
                conditions={Object {}}
                hasLastDeliveryWithAlert={false}
                trafficLightStatus="DELIVER"
              />
              <f1
                validate={true}
              >
                <styled.div>
                  <Unknown
                    date="2018-02-27T00:00:00.000Z"
                    message="comment"
                    title="Administrative Comments"
                  />
                </styled.div>
              </f1>
              <styled.div>
                <Button
                  accessibilityLabel=""
                  buttonStyle="primary"
                  disabled={false}
                  fontSize="0.875rem"
                  label="Acknowledge Patient Status"
                  loading={false}
                  minWidth=""
                  onClick={[Function]}
                />
              </styled.div>
            </Connect(Component)>
          </GridContainer>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Component)
              displayOnly={true}
              hideNextDeliveryDate={true}
            >
              <Unknown
                conditions={Object {}}
                hasLastDeliveryWithAlert={false}
                trafficLightStatus="DELIVER"
              />
              <f1
                validate={true}
              >
                <styled.div>
                  <Unknown
                    date="2018-02-27T00:00:00.000Z"
                    message="comment"
                    title="Administrative Comments"
                  />
                </styled.div>
              </f1>
              <styled.div>
                <Button
                  accessibilityLabel=""
                  buttonStyle="primary"
                  disabled={false}
                  fontSize="0.875rem"
                  label="Acknowledge Patient Status"
                  loading={false}
                  minWidth=""
                  onClick={[Function]}
                />
              </styled.div>
            </Connect(Component)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Unknown
                  conditions={Object {}}
                  hasLastDeliveryWithAlert={false}
                  trafficLightStatus="DELIVER"
                />,
                <f1
                  validate={true}
                >
                  <styled.div>
                    <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />
                  </styled.div>
                </f1>,
                <styled.div>
                  <Button
                    accessibilityLabel=""
                    buttonStyle="primary"
                    disabled={false}
                    fontSize="0.875rem"
                    label="Acknowledge Patient Status"
                    loading={false}
                    minWidth=""
                    onClick={[Function]}
                  />
                </styled.div>,
              ],
              "displayOnly": true,
              "hideNextDeliveryDate": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "conditions": Object {},
                  "hasLastDeliveryWithAlert": false,
                  "trafficLightStatus": "DELIVER",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": <styled.div>
                    <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />
                  </styled.div>,
                  "validate": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "date": "2018-02-27T00:00:00.000Z",
                      "message": "comment",
                      "title": "Administrative Comments",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    accessibilityLabel=""
                    buttonStyle="primary"
                    disabled={false}
                    fontSize="0.875rem"
                    label="Acknowledge Patient Status"
                    loading={false}
                    minWidth=""
                    onClick={[Function]}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "accessibilityLabel": "",
                    "buttonStyle": "primary",
                    "disabled": false,
                    "fontSize": "0.875rem",
                    "label": "Acknowledge Patient Status",
                    "loading": false,
                    "minWidth": "",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <LocalizedText
              fontSize={4}
              textKey="modals.patientStatusNotification.title"
            />
            <Styled(Styled(Component))
              onClick={[Function]}
            >
              <span>
                <Unknown
                  fillColor="#FFF"
                  height={14}
                />
              </span>
            </Styled(Styled(Component))>
          </styled.div>,
          <styled.div>
            <GridContainer>
              <Connect(Component)
                displayOnly={true}
                hideNextDeliveryDate={true}
              >
                <Unknown
                  conditions={Object {}}
                  hasLastDeliveryWithAlert={false}
                  trafficLightStatus="DELIVER"
                />
                <f1
                  validate={true}
                >
                  <styled.div>
                    <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />
                  </styled.div>
                </f1>
                <styled.div>
                  <Button
                    accessibilityLabel=""
                    buttonStyle="primary"
                    disabled={false}
                    fontSize="0.875rem"
                    label="Acknowledge Patient Status"
                    loading={false}
                    minWidth=""
                    onClick={[Function]}
                  />
                </styled.div>
              </Connect(Component)>
            </GridContainer>
          </styled.div>,
        ],
        "minWidth": 960,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <LocalizedText
                fontSize={4}
                textKey="modals.patientStatusNotification.title"
              />,
              <Styled(Styled(Component))
                onClick={[Function]}
              >
                <span>
                  <Unknown
                    fillColor="#FFF"
                    height={14}
                  />
                </span>
              </Styled(Styled(Component))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "fontSize": 4,
                "textKey": "modals.patientStatusNotification.title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  <Unknown
                    fillColor="#FFF"
                    height={14}
                  />
                </span>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <Unknown
                    fillColor="#FFF"
                    height={14}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "fillColor": "#FFF",
                    "height": 14,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <GridContainer>
              <Connect(Component)
                displayOnly={true}
                hideNextDeliveryDate={true}
              >
                <Unknown
                  conditions={Object {}}
                  hasLastDeliveryWithAlert={false}
                  trafficLightStatus="DELIVER"
                />
                <f1
                  validate={true}
                >
                  <styled.div>
                    <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />
                  </styled.div>
                </f1>
                <styled.div>
                  <Button
                    accessibilityLabel=""
                    buttonStyle="primary"
                    disabled={false}
                    fontSize="0.875rem"
                    label="Acknowledge Patient Status"
                    loading={false}
                    minWidth=""
                    onClick={[Function]}
                  />
                </styled.div>
              </Connect(Component)>
            </GridContainer>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Component)
                displayOnly={true}
                hideNextDeliveryDate={true}
              >
                <Unknown
                  conditions={Object {}}
                  hasLastDeliveryWithAlert={false}
                  trafficLightStatus="DELIVER"
                />
                <f1
                  validate={true}
                >
                  <styled.div>
                    <Unknown
                      date="2018-02-27T00:00:00.000Z"
                      message="comment"
                      title="Administrative Comments"
                    />
                  </styled.div>
                </f1>
                <styled.div>
                  <Button
                    accessibilityLabel=""
                    buttonStyle="primary"
                    disabled={false}
                    fontSize="0.875rem"
                    label="Acknowledge Patient Status"
                    loading={false}
                    minWidth=""
                    onClick={[Function]}
                  />
                </styled.div>
              </Connect(Component)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Unknown
                    conditions={Object {}}
                    hasLastDeliveryWithAlert={false}
                    trafficLightStatus="DELIVER"
                  />,
                  <f1
                    validate={true}
                  >
                    <styled.div>
                      <Unknown
                        date="2018-02-27T00:00:00.000Z"
                        message="comment"
                        title="Administrative Comments"
                      />
                    </styled.div>
                  </f1>,
                  <styled.div>
                    <Button
                      accessibilityLabel=""
                      buttonStyle="primary"
                      disabled={false}
                      fontSize="0.875rem"
                      label="Acknowledge Patient Status"
                      loading={false}
                      minWidth=""
                      onClick={[Function]}
                    />
                  </styled.div>,
                ],
                "displayOnly": true,
                "hideNextDeliveryDate": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "conditions": Object {},
                    "hasLastDeliveryWithAlert": false,
                    "trafficLightStatus": "DELIVER",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": <styled.div>
                      <Unknown
                        date="2018-02-27T00:00:00.000Z"
                        message="comment"
                        title="Administrative Comments"
                      />
                    </styled.div>,
                    "validate": true,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Unknown
                        date="2018-02-27T00:00:00.000Z"
                        message="comment"
                        title="Administrative Comments"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "date": "2018-02-27T00:00:00.000Z",
                        "message": "comment",
                        "title": "Administrative Comments",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      accessibilityLabel=""
                      buttonStyle="primary"
                      disabled={false}
                      fontSize="0.875rem"
                      label="Acknowledge Patient Status"
                      loading={false}
                      minWidth=""
                      onClick={[Function]}
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "accessibilityLabel": "",
                      "buttonStyle": "primary",
                      "disabled": false,
                      "fontSize": "0.875rem",
                      "label": "Acknowledge Patient Status",
                      "loading": false,
                      "minWidth": "",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
